extensions/vscode/src/activation/activate.ts

```diff
@@ -52,14 +52,19 @@ export async function activateExtension(context: vscode.ExtensionContext) {
   }
 
   const api = new VsCodeContinueApi(vscodeExtension);
-  const showOnboarding = !isGraniteOnboardingComplete(context);
-  vscode.commands.executeCommand(
+  const graniteOnboardingComplete = isGraniteOnboardingComplete(context);
+  await vscode.commands.executeCommand(
     "setContext",
     "granite.initialized",
-    !showOnboarding,
+    graniteOnboardingComplete,
   );
-  if (showOnboarding) {
-    await vscode.commands.executeCommand("granite.setup");
+
+  if (!graniteOnboardingComplete) {
+    const showOnboardingAtStartup = context.globalState.get("initialSetup", true);
+    if (showOnboardingAtStartup) {
+      await context.globalState.update("initialSetup", !showOnboardingAtStartup);
```

Comment:
Can you make this another constant in `core/granite/commons/constants.ts`? It's weird to me to have two things in globalState that are uncoordinated and in different places. Would be good to briefly (one-line) document their meaning - we can't rename `showGraniteOnboardingCard` but it doesn't really mean what it says any more.

Also, I think if graniteOnboardingComplete is set we should *still* clear the initialSetup variable.

Reply:
done (I think ;-))